pkg/eval/vals/equal.go

```diff
@@ -22,6 +23,12 @@ func Equal(x, y interface{}) bool {
 	case bool:
 		return x == y
 	case float64:
+		if math.IsNaN(x) && math.IsNaN(y.(float64)) {
+			// `NaN == NaN` is always false since NaN is never equal to any
+			// other value; not even NaN. But for unit tests we want to ensure
+			// that if the test is expected to produce NaN it does so.
+			return true
```

Comment:
Ugh! I just realized this changes the behavior of `eq (float64 nan) (float64 nan)` but not `== (float64 nan) (float64 nan)`. The `eq` documentation says it checks if the values `are structurally equivalent` which isn't rigorously defined. Ideally this redefinition of NaN equality would only apply to the unit test framework and not affect `eq`. Nonetheless, it's not clear that the current behavior is a problem. In fact, it might be considered a feature; albeit one that needs to be documented.